src/core/TriangleSplitter.js

```diff
@@ -189,7 +221,28 @@ export class TriangleSplitter {
 				// so we can use that information to determine whether to split later.
 				const startDist = plane.distanceToPoint( _edge.start );
 				const endDist = plane.distanceToPoint( _edge.end );
-				if ( Math.abs( startDist ) < COPLANAR_EPSILON && Math.abs( endDist ) < COPLANAR_EPSILON ) {
+				let coPlanarEpsilonStart = COPLANAR_EPSILON *	Math.max( 1, 0.5 * _edge.start.distanceTo( planeCenter ) / planeEdgeSize );
+				let coPlanarEpsilonEnd = COPLANAR_EPSILON *	Math.max( 1, 0.5 * _edge.end.distanceTo( planeCenter ) / planeEdgeSize );
+
+				if ( coPlanarEpsilonStart > COPLANAR_EPSILON_MAX ) {
+
+					coPlanarEpsilonStart = COPLANAR_EPSILON_MAX;
+					console.warn( 'High coplanar epsilon start because of degenerate faces' );
+
+				}
+
+				if ( coPlanarEpsilonEnd > COPLANAR_EPSILON_MAX ) {
+
+					coPlanarEpsilonEnd = COPLANAR_EPSILON_MAX;
+					console.warn( 'High coplanar epsilon end because of degenerate faces' );
+
+				}
```

Comment:
These logs seem risky to me. This function gets called a lot and logs slow down performance significantly. Have you seen these get output a lot while testing? I'm thinking it might be best to remove the logs.

Reply:
In usual cases, it does not log a lot or it does not log. But in degenerate cases (all coplanar faces, many CSG operations called sequentially it can log a lot.

Reply:
Yeah I think if anything we should debounce the print statement so we can avoid flooding the log.

Reply:
It is done